rn/Teacher/ios/Submissions/HideGradesViewController.swift

```diff
@@ -0,0 +1,110 @@
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var hideGradesButton: DynamicButton!
+    private var sections: [String] = []
+    private var sectionToggles: [Bool] = []
+    private var gradesCurrentlyPosted = 0
+    private var showSections = false
+    @IBOutlet weak var allGradesHiddenView: UIView!
+    @IBOutlet weak var allHiddenLabel: DynamicLabel!
+    @IBOutlet weak var allHiddenSubHeader: DynamicLabel!
+
+    static func create() -> HideGradesViewController {
+        let controller = loadFromStoryboard()
+        return controller
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setupSections()
+
+        allGradesHiddenView.isHidden = true
+        allHiddenLabel.text = NSLocalizedString("All Hidden", comment: "")
+        allHiddenSubHeader.text = NSLocalizedString("All grades are currently hidden.", comment: "")
+
+        hideGradesButton.setTitle(NSLocalizedString("Hide Grades", comment: ""), for: .normal)
+    }
+
+    func setupTableView() {
+        tableView.register(PostGradesViewController.SectionCell.self, forCellReuseIdentifier: String(describing: PostGradesViewController.SectionCell.self))
+    }
+
+    func setupSections() {
+        sections = ["section A", "section B"]     // FIXME: remove these static sections once we have real sections
+        sectionToggles = Array(repeating: false, count: sections.count)
+    }
+
+    @IBAction func actionUserDidClickHideGrades(_ sender: Any) {
+        print(#function)
+    }
+}
+
+extension HideGradesViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return (showSections ? sections.count : 0) + 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: PostGradesViewController.SectionCell = tableView.dequeue(for: indexPath)
+
+        if indexPath.row == 0 {
+            cell.textLabel?.text = NSLocalizedString("Specific Sections", comment: "")
+            cell.selectionStyle = .none
+            cell.toggle.removeTarget(self, action: #selector(actionDidToggleShowSections(sender:)), for: UIControl.Event.valueChanged)
+            cell.toggle.isOn = showSections
+            cell.toggle.addTarget(self, action: #selector(actionDidToggleShowSections(sender:)), for: UIControl.Event.valueChanged)
+        } else {    //  sections
+            let index = abs(indexPath.row - 1)
+            cell.textLabel?.text = sections[index]
+            cell.selectionStyle = .none
+            cell.toggle.removeTarget(self, action: #selector(actionDidToggleSection(toggle:)), for: UIControl.Event.valueChanged)
+            cell.toggle.isOn = sectionToggles[index]
+            cell.toggle.tag = index
+            cell.toggle.addTarget(self, action: #selector(actionDidToggleSection(toggle:)), for: UIControl.Event.valueChanged)
```

Comment:
why remove and then add the same target? is that to prevent duplicates?

Reply:
removed